realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1066,7 +1066,7 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFindWithHandover
 {
     TR_ENTER()
     try {
-        std::unique_ptr<Query> query = getHandoverQuery(bgSharedGroupPtr, replicationPtr, queryPtr);
+        std::unique_ptr<Query> query = getHandoverQuery(bgSharedGroupPtr, replicationPtr, queryPtr, false);
```

Comment:
Remind me again: What is the reason that findFirst have to behave differently than findAll(). This is also evident in HandlerController where the flow is hard to follow because AsyncRealmObjects behave differently than AsyncRealmResults?